src/WebJobs.Script.WebHost/Configuration/HostHstsOptionsSetup.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Script.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
+{
+    internal class HostHstsOptionsSetup : IConfigureOptions<HostHstsOptions>
+    {
+        private readonly IConfiguration _configuration;
+
+        public HostHstsOptionsSetup(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void Configure(HostHstsOptions options)
+        {
+            IConfigurationSection jobHostSection = _configuration.GetSection(ConfigurationSectionNames.JobHost);
+            var hstsSection = jobHostSection.GetSection(ConfigurationSectionNames.Hsts);
+
+            if (hstsSection.Exists())
```

Comment:
I think I made this comment in the other PR. But Can't we just call `Bind()` here? If the value is there, it'll apply it; otherwise it won't. I don't think we do these kinds of checks anywhere else.

Reply:
We can do that as well, `hstsSection.Exists()` is used to make `IsEnabled` property mandatory. We can just call `Bind()` here and it would automatically apply hsts config only if `isEnabled` is set to true otherwise it would be a no-op. 

I added this explicitly because this is a security feature. [Hsts options ](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.httpspolicy.hstsoptions?view=aspnetcore-2.2#properties) is documented under aspnet core and probably widely used. The users could end up adding the hsts config without setting `isEnabled` assuming that the configuration is applied while in reality it isn't because `isEnabled` is not set. We do enforce configuration in a similar manner in a couple of places at `ScriptHostOptionsSetup` and `ExtensionBundleConfigurationHelper` however I don't have any strong inclinations of doing it one way or other.